maltoolbox/attackgraph/attackgraph.py

```diff
@@ -576,42 +599,67 @@ def _generate_graph(self) -> None:
+
+            lang_graph_attack_step = lang_graph_asset.attack_steps[\
+                ag_node.name]
+
+            while lang_graph_attack_step:
+                for child in lang_graph_attack_step.children:
+                    for target_attack_step, expr_chain in \
+                            lang_graph_attack_step.children[child]:
+                        target_assets = self._follow_expr_chain(
+                            self.model,
+                            [ag_node.asset],
+                            expr_chain
                         )
-                    ag_node.children.append(target_node)
-                    target_node.parents.append(ag_node)
+
+                        for target_asset in target_assets:
+                            if target_asset is not None:
+                                target_node_full_name = target_asset.name + \
+                                    ':' + target_attack_step.name
+                                target_node = self.get_node_by_full_name(
+                                    target_node_full_name)
+                                if target_node is None:
+                                    msg = ('Failed to find target node '
+                                           '"%s" to link with for attack '
+                                           'step "%s"(%d)!')
+                                    logger.error(
+                                        msg,
+                                        target_node_full_name,
+                                        ag_node.full_name,
+                                        ag_node.id
+                                    )
+                                    raise AttackGraphStepExpressionError(
+                                        msg % (
+                                            target_node_full_name,
+                                            ag_node.full_name,
+                                            ag_node.id
+                                        )
+                                    )
+
+                                assert ag_node.id is not None
+                                assert target_node.id is not None
+
+                                logger.debug('Linking attack step "%s"(%d) '
+                                    'to attack step "%s"(%d)' %
+                                    (
+                                        ag_node.full_name,
+                                        ag_node.id,
+                                        target_node.full_name,
+                                        target_node.id
+                                    )
+                                )
+                                ag_node.children.append(target_node)
+                                target_node.parents.append(ag_node)
+                if bool(lang_graph_attack_step.overrides):
+                    break
+                lang_graph_attack_step = lang_graph_attack_step.inherits
```

Comment:
Why do you need to cast to `bool`? Isn't it already a bool? 

```suggestion
                if not lang_graph_attack_step.overrides:
                    lang_graph_attack_step = lang_graph_attack_step.inherits
```

Reply:
The comment is correct, it is already a bool, it doesn't need the cast. The code suggestion is not correct, this is a while loop that iterates through the language graph attack steps inheritance hierarchy.

If it ever runs into an overriden step it needs to exit that loop. Your suggestion doesn't do that which leads to an infinite loop if it does have an overriden step because the `lang_graph_attack_step` will no longer change and the loop does not break.